src/odemis/acq/stream/_sync.py

```diff
@@ -1140,46 +1143,55 @@ def _runAcquisitionEbeam(self, future):
 
-        # get the metadata (should be the same for all images)
-        md_sem = self._acq_data[0][-1].metadata
-        md_sem[model.MD_INTEGRATION_COUNT] = n
+                    # get the metadata from last image (should be the same for all images)
+                    md_sem = self._acq_data[stream_idx][-1].metadata
+                    md_sem[model.MD_DWELL_TIME] *= n  # total time ebeam stayed on same pixel/position
+                    md_sem[model.MD_INTEGRATION_COUNT] = n
 
-        # delete the images which need to be averaged and replace
-        del self._acq_data[0][-n:]
+                    # replace images, which needed averaging, with averaged image
+                    self._acq_data[stream_idx][-n:] = [model.DataArray(data_sem, md_sem)]
 
-        self._acq_data[0].append(model.DataArray(data_sem, md_sem))
+                elif det_type == model.MD_DT_INTEGRATING:  # optical
+                    # get the images, which need to be integrated
+                    data2integrate = self._acq_data[stream_idx][-n:]
 
-        # get the images, which need to be integrated
-        data2integrate = self._acq_data[self._ccd_idx][-n:]
+                    # get the metadata from last image (should be the same for all images)
+                    md_optical = self._acq_data[stream_idx][-1].metadata
+                    md_optical[model.MD_EXP_TIME] *= n
+                    md_optical[model.MD_INTEGRATION_COUNT] = n
 
-        # get the metadata (should be the same for all images)
-        md_optical = self._acq_data[self._ccd_idx][-1].metadata
-        if hasVA(self._sccd, "exposureTime"):
-            md_optical[model.MD_EXP_TIME] = self._sccd.exposureTime.value
-        elif hasVA(self._sccd, "integrationTime"):
-            md_optical[model.MD_EXP_TIME] = self._sccd.integrationTime.value
-        md_optical[model.MD_INTEGRATION_COUNT] = n
-        # delete the images which need to be integrated and replace with integrated image at end of method
-        del self._acq_data[1][-n:]
+                    # logging.debug("Integrate %s images", n)
+                    orig_dtype = data2integrate[0].dtype
+                    best_dtype = get_best_dtype_for_acc(orig_dtype, n)  # avoid saturation and overflow
+                    data = numpy.sum(data2integrate, axis=0, dtype=best_dtype)
 
-        logging.debug("Integrate %s images", n)
-        orig_dtype = data2integrate[0].dtype
-        best_dtype = get_best_dtype_for_acc(orig_dtype, n)  # avoid saturation and overflow
-        data = numpy.sum(data2integrate, axis=0, dtype=best_dtype)
+                    if model.MD_BASELINE in md_optical:
+                        baseline = md_optical[model.MD_BASELINE]
+                        # Subtract baseline (aka black level) to avoid it from being multiplied
+                        # remove baseline from n-1 images, keep one baseline as bg level
+                        data = data - (n - 1) * baseline
```

Comment:
Yes... but maybe first double check that you're not going to get negative values, because they would underflow. So I'd suggest to use the same code as in`driver.spectrometer`.